class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -1,110 +1,94 @@
 #include <string>
-#include <vector>
 #include <iostream>
 #include <fstream>
 #include <sstream>
+#include <map>
 
-std::vector<std::string> splitString(const std::string& str, char delimiter) {
-    std::vector<std::string> tokens;
-    std::stringstream ss(str);
-    std::string token;
-
-    while (std::getline(ss, token, delimiter)) {
-        tokens.push_back(token);
-    }
-    return tokens;
+namespace {
+    const std::string bread = "bread";
+    const std::string deleteCommand = "delete";
 }
-int main(int argc, char* argv[])
-{
-    std::string argument2;
-        
-    std::string WordToFind;
-    if (argc < 2) {
-        std::cout << "Please provide a word to find." << std::endl;
-        return 0;
-    }
-    WordToFind= argv[1];
-    if (argc>2)
-    {
-        argument2 = argv[2];
+class Database {
+private:
+    std::map<std::string, int> statsData;
+    std::string database;
+    void LoadData() {
+        std::ifstream file(database);
+        std::string name;
+        int count;
+        while (file >> name >> count) {
+            statsData [name] = count;
+        }
+        file.close();
     }
-    if (WordToFind=="bread")
-    {
-        std::ofstream file("database.txt", std::ios::trunc); 
+    void SaveData() {
+        std::ofstream file(database, std::ios::trunc);
+        for (const auto& [name, count] : statsData ) {
+            file << name << " " << count << std::endl;
+        }
         file.close();
-        std::cout << "all data was cleared" << std::endl;
-        return 0;
     }
-    std::ifstream file("database.txt");
-    std::string line;
-    int lineNumber = 0;
-    bool found = false;
-    std::string lineFound;
-
-    std::vector<std::string> lines;
- 
-    while (std::getline(file, line)) {
-        lines.push_back(line);
+ std::string deleteUser(const std::string& nameToFind) {
+    auto name = statsData.find(nameToFind);
+    if (name != statsData.end()) {
+        statsData.erase(name);
     }
-    file.close();
-    for (const std::string& line : lines) {
-        lineNumber++;
-        if (line != "") {
-            if (splitString(line, ' ')[0] == WordToFind)
-            {
-                found = true;
-                lineFound = line;
-                break;
-            }
+    SaveData();
+    return "All data about this user was cleared";
```

Comment:
1. let's return bool instead of string to split data layer and business logic.
2. Also we can avoid unneeded SaveData call if database was not changed. And nameIt instead of name - it is iterator let's use proper name then
```
auto nameIt = ...;
if (nameIt != statsData.end()) {
      statsData.erase(nameIt);
      SaveData();
      return true;
}

return false;
```